packages/kotti-navbar/src/KtNavbarLink.vue

```diff
@@ -0,0 +1,29 @@
+<template>
+	<div class="navbar-link">
+		<a
+			v-if="isExternal"
+			@click="$emit('menuLinkClick', link)"
+			v-bind="$attrs"
+			:href="link"
+		>
+			<slot />
+		</a>
+		<router-link
+			v-else
+			@click="$emit('menuLinkClick', link)"
+			v-bind="$attrs"
+			active-class="is-active"
+			:to="link"
+		>
+			<slot />
+		</router-link>
```

Comment:
We prefer to not have a peer dependency if possible, so consider implementing a scooped slot instead or the dynamic component element 

```
<component 
  :is="linkComponent"   // defaults to 'a'
  @click="$emit('menuLinkClick', link)"
  v-bind="$attrs"
  active-class="is-active"
  :to="link"
  :href="link"
/>
```

Reply:
If you want to allow KtNavBarLink to consume state from the Navbar while building customizable implementations of it then you can use the provider/inject pattern to allow users the ability to create and reuse their own custom NabarLink component that will work with the KtNavbar by wrapping it

Reply:
the dynamic component suggestion I made is not completely thought through and may need reworking (in case we're passing the a tag too many props